### PR TITLE
[Concurrency] repair cooperative global executor

### PR DESCRIPTION
**Pull request by kateinoigakukun**
`SWIFT_STDLIB_SINGLE_THREADED_RUNTIME` mode has been broken for a long time.

This patch guards some includes and use of libdispatch headers so that platforms
that doesn't support libdispatch can build cooperative executor runtime.
And fixed missing implementations for cooperative mode.

CC: @MaxDesiatov 

Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves SR-NNNN.

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test

**Review by ktoso — COMMENTED** [review]
LGTM, but I wonder if we should add some way to test if this mode works so we don't keep breaking it on accident?

Do we have any way to test such thing @DougGregor ?

**Review by ktoso — APPROVED** [review]

**Comment by MaxDesiatov** [comment]
@ktoso should the lack of tests block this PR from being merged? I'm not sure if there's any other way to test this other than adding Swift for WebAssembly target to the community CI build matrix. This "build configuration" can't be enabled with a driver flag I suppose, and we can't rebuild with different CMake flags on existing platforms, right?

**Review by rjmccall — APPROVED** [review]

**Comment by rjmccall** [comment]
I have no problem landing this without testing in the short term, but yeah, I think it should be a goal to bring up CI to test the WebAssembly target ASAP.

**Pull request merged**
